infrastructure/ansible/roles/http_server/tasks/main.yml

```diff
@@ -44,8 +46,17 @@
     daemon_reload: yes
 
 - name: enable and start service
+  when: (deploy_jar_file.changed == false) and (deploy_config_file.changed == false)
   become: true
   service:
     name: http_server
     state: started
     enabled: yes
+
+- name: restart service if new jar file deployed
+  when: (deploy_jar_file.changed) or (deploy_config_file.changed)
+  become: true
+  service:
```

Comment:
You might want to consider using `systemd` explicitly, see https://stackoverflow.com/a/44099145

Reply:
Agree it's a good modernizing move.